src/production/transformation/processTransformation.ts

```diff
@@ -21,7 +21,7 @@ function onGlobal(node: Identifier, props: LocalContext) {
     }
     if (!props.globalInserted) {
       props.globalInserted = true;
-      props.insertions.push(`const global = {};`);
+      props.insertions.push(`const global = globalThis || self || window || global;`);
```

Comment:
you actually need to check them for being undefined. Otherwise, it will throw an error
```
typeof window !== "undefined" ? window : ...
```

Reply:
Why will it throw an error?

This works:
```js
let a = undefined
let b = undefined
let c = { prop: 6 }
let d = { prop2: 12 }
let g = a || b || c || d
// g = c
```

Reply:
`let g = a || b || c || d`

this will throw ^ Since those variables are not defined

Reply:
You can  open a debugger a type `global` you will see what I mean. Either way, I would just assign global to window if defined. If I am not mistaken that is being applied only for browser builds?

Reply:
Oh yeah, I don't know what I was thinking.

Reply:
Regarding your question, NodeJS has `global` and in workers you can't use `window` only `self`. So I think the order is good:
`globalThis` else `self` else `window` else `global`